android/java/org/chromium/chrome/browser/rewards/BraveRewardsCustomTipFragment.java

```diff
@@ -0,0 +1,181 @@
+            }
+        });
+    }
+
+    @SuppressLint("SetTextI18n")
+    private void updateText(View view) {
+        exchangeRate = BraveRewardsNativeWorker.getInstance().GetWalletRate();
+        currencyOneEditText = view.findViewById(R.id.currencyOneEditText);
+        currencyTwoTextView = view.findViewById(R.id.currencyTwoTextView);
+        currencyOneEditText.setFilters(new InputFilter[] {new DecimalDigitsInputFilter(5, 2)});
+        currencyOneEditText.addTextChangedListener(textChangeListener);
+        mSelectedAmount = getArguments().getDouble(SELECTED_AMOUNT);
+        currencyOneEditText.setText(String.valueOf(mSelectedAmount)); // Default value
+        ((BraveRewardsSiteBannerActivity) getActivity())
+                .onAmountChange(mSelectedAmount, roundExchangeUp(mSelectedAmount * exchangeRate));
+    }
+
+    private TextWatcher textChangeListener = new TextWatcher() {
+        @Override
+        public void beforeTextChanged(CharSequence s, int start, int count, int after) {}
+
+        @SuppressLint("SetTextI18n")
+        @Override
+        public void onTextChanged(CharSequence s, int start, int before, int count) {
+            if (s == null || s.toString().isEmpty()) s = "0";
+            Double batValue = getBatValue(s.toString());
+            Double usdValue = exchangeRate * batValue;
+
+            ((BraveRewardsSiteBannerActivity) getActivity())
+                    .onAmountChange(batValue, roundExchangeUp(usdValue));
+            if (isBatCurrencyMode) {
+                currencyTwoTextView.setText(String.valueOf(roundExchangeUp(usdValue)));
+            } else {
+                currencyTwoTextView.setText(String.valueOf(batValue));
+            }
+        }
+
+        @Override
+        public void afterTextChanged(Editable s) {}
+    };
+
+    class DecimalDigitsInputFilter implements InputFilter {
+        private Pattern mPattern;
+
+        DecimalDigitsInputFilter(int digitsBeforeZero, int digitsAfterZero) {
+            mPattern = Pattern.compile("[0-9]{0," + (digitsBeforeZero - 1) + "}+((\\.[0-9]{0,"
+                    + (digitsAfterZero - 1) + "})?)||(\\.)?");
+        }
+
+        @Override
+        public CharSequence filter(
+                CharSequence source, int start, int end, Spanned dest, int dstart, int dend) {
+            Matcher matcher = mPattern.matcher(dest);
+            if (!matcher.matches()) return "";
+            return null;
+        }
+    }
+
+    /**
+     *
+     * @param inputValue : editText string passed here
+     * @return
+     * convert to bat value if current editText is usd
+     * It will return multiple 0.25
+     * And also return which is lower between Max and value
+     * decimal points always roundedOff to floor value.
+     */
+    private double getBatValue(String inputValue) {
+        double rawValue = Double.parseDouble(inputValue);
+        if (!isBatCurrencyMode) {
+            // from usd to BAT
```

Comment:
nit: `usd` -> `USD` and same `editText is usd`

Reply:
updated